nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -499,4 +507,20 @@ protected boolean checkEntryTenantId(String entryTenantId) {
         return true;
     }
 
+    protected String getPrefixedIdField() {
+        Field idField = directory.getSchemaFieldMap().get(getIdField());
+        if (idField == null) {
+            return null;
+        }
+        return idField.getName().getPrefixedName();
+    }
+
+    protected String getPrefixedPasswordField() {
```

Comment:
You could have factored out the body of `getPrefixedIdField` and `getPrefixedPasswordField` which are very similar.